packages/babel-parser/src/parser/expression.js

```diff
@@ -704,11 +712,11 @@ export default class ExpressionParser extends LValParser {
 
   atPossibleAsync(base: N.Expression): boolean {
     return (
-      !this.state.containsEsc &&
```

Comment:
`containsEsc` only says if the current token has escape sequences. This references the token after `async` which is the wrong one.